apiserver/facades/agent/machine/machiner.go

```diff
@@ -41,6 +42,9 @@ type NetworkService interface {
 // MachineService defines the methods that the facade assumes from the Machine
 // service.
 type MachineService interface {
+	// WatchMachines returns a StringsWatcher that notifies of the changes
+	// in the machines table for the model.
+	WatchMachines(context.Context) (watcher.StringsWatcher, error)
```

Comment:
This is not used by the machiner. The machiner only watches a single machine.

Reply:
This was because the `DeadEnsurer` and the `ModelMachinesWatcher` are both in the common package, sharing a common `MachineService` interface. 

Fixing now so they use individual interfaces to get only the required methods 👍